app/src/main/java/com/github/se/eduverse/ui/search/UserProfileScreen.kt

```diff
@@ -0,0 +1,261 @@
+                      currentUserId = FirebaseAuth.getInstance().currentUser?.uid ?: "",
+                      profileViewModel = viewModel,
+                      onPublicationClick = { /* Handle publication click */},
+                      modifier = Modifier.testTag("publications_grid"))
+                }
+              }
+            }
+      }
+}
+
+@Composable
+private fun ErrorMessage(message: String, modifier: Modifier = Modifier) {
+  Box(modifier = modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+    Text(
+        text = message,
+        style = MaterialTheme.typography.bodyLarge,
+        color = MaterialTheme.colorScheme.error,
+        modifier = Modifier.testTag("error_message"))
+  }
+}
+
+@Composable
+private fun StatItem(label: String, count: Int, modifier: Modifier = Modifier) {
+  Column(modifier = modifier, horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(
+        text = count.toString(),
+        style = MaterialTheme.typography.titleMedium,
+        modifier = Modifier.testTag("stat_count_$label"))
+    Text(
+        text = label,
+        style = MaterialTheme.typography.bodyMedium,
+        color = MaterialTheme.colorScheme.onSurfaceVariant,
+        modifier = Modifier.testTag("stat_label_$label"))
+  }
+}
+
+@Composable
+private fun PublicationsGrid(
+    publications: List<Publication>,
+    currentUserId: String,
+    profileViewModel: ProfileViewModel,
+    onPublicationClick: (Publication) -> Unit,
+    modifier: Modifier = Modifier
+) {
+  var selectedPublication by remember { mutableStateOf<Publication?>(null) }
+
+  if (publications.isEmpty()) {
+    Box(
+        modifier = Modifier.testTag("empty_publications_container").fillMaxSize(),
+        contentAlignment = Alignment.Center) {
+          Text(
+              text = "No publications yet",
+              style = MaterialTheme.typography.bodyLarge,
+              color = MaterialTheme.colorScheme.onSurfaceVariant,
+              modifier = Modifier.testTag("empty_publications_text"))
+        }
+  } else {
+    LazyVerticalGrid(
+        columns = GridCells.Fixed(3),
+        modifier = modifier.fillMaxSize(),
+        contentPadding = PaddingValues(1.dp),
+        horizontalArrangement = Arrangement.spacedBy(1.dp),
+        verticalArrangement = Arrangement.spacedBy(1.dp)) {
+          items(publications) { publication ->
+            PublicationItem(
+                publication = publication,
+                onClick = { selectedPublication = publication },
+                modifier = Modifier.testTag("publication_item_${publication.id}"))
+          }
+        }
+
+    // Show detail dialog when a publication is selected
+    selectedPublication?.let { publication ->
+      PublicationDetailDialog(
+          publication = publication,
+          profileViewModel = profileViewModel,
+          currentUserId = currentUserId,
+          onDismiss = { selectedPublication = null })
+    }
```

Comment:
This method is the same as in ProfileScreen, you should use only one in both case (you can put it in file `Helper`)

Reply:
i tried to do that but it was a mess because of the mutable state inside of it, so i just dupilcated it but i will try to solve this dupliation and refactor the code in a future PR.

Reply:
Ok